### ISSUE TITLE
ADAM Files Not Inheriting Permissions in Secure Folders

### ISSUE THREAD

**Issue by mryckman** [2sic/2sxc#1430]
**I'm submitting a ...**  
[x] bug report 

**...about**   
[x] other / unknown

**Current behavior**
We set our "Content" folder to be a "secure" folder in DNN (which should probably be the default) and have updated templates in 2sxc to accommodate file lookups. Until an update in the past six months or so (sorry... I'm not sure which), files would automatically inherit their parent folders' permissions. For us, we had all content files accessible to "Registered Users". So, this meant that all files in the Adam/Content folder were given access to "Registered Users". 

Recently, all files added seem to be given admin only permissions and so we have to go to the root content folder and copy all permissions down.

**Expected behavior**
Files, and their new parent folders, should inherit the permissions of the parent "Content" folder.

**Minimal reproduction of the problem with instructions**
On a new install, make a secure "content" folder and restrict the permissions on the content folder by role.

**What is the motivation / use case for changing the behavior?**
ADAM files are insecure and impossible to manage without using a secure or database folder type. We needed to do it this way to make 2sxc usable for our intranet.

**Please tell us about your environment:**

* **2sxc version(s):** 9.10.0  
* **Browser:** [all]
* **DNN:** [9.1 ] 
* **Language:** [ English ]

**Anything you would like to add**

**Comment by iJungleboy** [comment]
AFAIK we never touched / changed this. Could this also be a (probably accidental) dnn behavior of dnn 9?

**Comment by iJungleboy** [comment]
Could you re-check that it's not a changed dnn version behavior, otherwise we won't follow up on it.

**Comment by mryckman** [comment]
Sorry for the delay. I checked this and I think it's a 2sxc problem. The parent folder is receiving the inherited permissions, but the "files" folder is not. For example, when an entity was created, a folder with the hash key (or whatever that is) was added. That folder received the inherited permissions. But, the "Files" folder inside didn't.

I looked into the tables and I might see what's happening. The creation time for the first entry of permissions on the parent folder start at 2017-11-27 09:26:19.983. For that folder, there are 11 more permissions entries that end at 2017-11-27 09:26;20.077. In other words, it took a few milliseconds to add the permissions to the parent. 

But, the permissions for the "Files" folder - the child - there's only one entry and it starts at 2017-11-27 09:26:19.993. 

In other words, it looks like both folders were added to quickly that the permissions for the parent weren't fully in place before the child was added. So, the child probably never picked them up. This would also explain why the error has been intermittent.

What do you think?
